actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/org/impl/OrganisationClientImpl.java

```diff
@@ -67,20 +69,27 @@ private String upsertOrg(
     request.setRequest(orgMap);
     request.setOperation(operation);
     request.getContext().put(JsonKey.CALLER_ID, JsonKey.BULK_ORG_UPLOAD);
-    Object obj = interServiceCommunication.getResponse(actorRef, request);
-
-    if (obj instanceof Response) {
-      Response response = (Response) obj;
-      orgId = (String) response.get(JsonKey.ORGANISATION_ID);
-    } else if (obj instanceof ProjectCommonException) {
-      throw (ProjectCommonException) obj;
-    } else if (obj instanceof Exception) {
-      throw new ProjectCommonException(
-          ResponseCode.SERVER_ERROR.getErrorCode(),
-          ResponseCode.SERVER_ERROR.getErrorMessage(),
-          ResponseCode.SERVER_ERROR.getResponseCode());
+    try {
+      Timeout t = new Timeout(Duration.create(10, TimeUnit.SECONDS));
+      Future<Object> future = Patterns.ask(actorRef, request, t);
+      Object obj = Await.result(future, t.duration());
+
+      if (obj instanceof Response) {
+        Response response = (Response) obj;
+        orgId = (String) response.get(JsonKey.ORGANISATION_ID);
+      } else if (obj instanceof ProjectCommonException) {
```

Comment:
else if and else condition not required

Reply:
Here we are checking the response obj type and throwing the exception, so thinking to keep it for now.